### PR TITLE
Add taxes endpoint and data store

### PR DESCRIPTION
**Pull request by joshuatf**
Fixes (partially) #844 

Adds in the endpoint for the taxes report table.

### Screenshots
screen shot 2018-12-07 at 8 56 45 am

### Detailed test instructions:

1.  Add at least a couple of tax rates under `/wp-admin/admin.php?page=wc-settings&tab=tax&section=standard`
2.  Add several rows of data into the `wp_wc_order_tax_lookup` table.  This can be any data, but match the tax rate ID to an existing tax rate from step 1 for more thorough testing.
3.  Make a post request to `wp/wp-json/wc/v3/reports/taxes` with WC Rest API keys.
4.  Make sure you receive accurate data from the request for these fields:
    * tax_rate_id
    * name
    * country
    * state
    * priority
    * total_tax
    * order_tax
    * shipping_tax
    * orders_count
    * _links
5.  Test with these query params as well
    * before
    * after
    * page
    * per_page
    * orderby
    * order
    * taxes (comma separated tax_rate_ids from the `woocommerce_tax_rates` table
6. Run phpunit tests to make sure they pass.

### Todo
This PR is already very large, so it's being broken down into smaller pieces.
* Tax report hookup
* Tax report stats
* Tax report stats hookup
* `wp_wc_order_tax_lookup` syncing

### CODE REVIEW

File: includes/data-stores/class-wc-admin-reports-taxes-data-store.php [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+
+			$tax_data = $wpdb->get_results(
+				"SELECT
+						{$selections}
+					FROM
+						{$table_name}
+						{$sql_query_params['from_clause']}
+					WHERE
+						1=1
+						{$sql_query_params['where_clause']}
```
Comment (jeffstieler) [inline includes/data-stores/class-wc-admin-reports-taxes-data-store.php]: Looks like this is missing `where_time_clause`. (Found testing `before`/`after` query params).

File: includes/data-stores/class-wc-admin-reports-taxes-data-store.php [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+		'orders_count' => 'COUNT(DISTINCT order_id) as orders_count',
+	);
+
+	/**
+	 * Constructor
+	 */
+	public function __construct() {
+		global $wpdb;
+		$table_name                          = $wpdb->prefix . self::TABLE_NAME;
+		$this->report_columns['tax_rate_id'] = $table_name . '.' . $this->report_columns['tax_rate_id'];
```
Comment (jeffstieler) [inline includes/data-stores/class-wc-admin-reports-taxes-data-store.php]: Curious - why is this needed?

**Review by psealock — COMMENTED** [review]
I just came across a bug which may be outside the scope of this PR but renders the `total_tax` and `order_tax` portion of the response incorrectly.

I created two orders that would involve a tax rate, but it appears the `wp_wc_order_tax_lookup` table was populated by many duplicate rows:

screen shot 2018-12-10 at 1 10 22 pm

When I hit the endpoint, the response sums up the duplications

screen shot 2018-12-10 at 1 10 34 pm

It should be

```
"total_tax": 91.6,
"order_tax": 91.6,
```

**Comment by joshuatf** [comment]
@psealock I also had issues with the `wc_order_tax_lookup` table not populating correctly, but in my case no information gets added to the table at all.

Opened a separate issue here - https://github.com/woocommerce/wc-admin/issues/1009

File: includes/data-stores/class-wc-admin-reports-taxes-data-store.php (hunk 1 above)
Comment (joshuatf) [reply includes/data-stores/class-wc-admin-reports-taxes-data-store.php]: Nice, catch @jeffikus !  Added this and it's testing well for me.  Can you give it another shot?

**Review by jeffstieler — APPROVED** [review]
